Sources/zcash-swift-payment-uri/Parser.swift

```diff
@@ -0,0 +1,427 @@
+//
+//  Parser.swift
+//
+//
+//  Created by Francisco Gindre on 2023-13-08.
+//
+
+import Foundation
+import Parsing
+
+/// Result of the URI parser. This returns a `PaymentRequest` if a compliant
+/// [ZIP-321](https://zips.z.cash/zip-0321) request is parsed from
+/// a URI String. This implementation accounts of legacy schemes like that preceded
```

Comment:
thanks for clarifying this point @nuttycom.

I guess we need to rework this section 

> Backward compatibility 
As this ZIP is written, several clients already implement a zcash: URI scheme similar to this one, however usually without the additional req- prefix requirement or the facility to specify multiple payments using paramindex. These implementations also generally do not support URIs, even with a single payment, where the address is specified as an address= query parameter rather than in the hier-part of the URI. They may also not support the memo parameter, or may not treat it as base64url-encoded.


because then it's unclear to me as reader what should I do with this contextual information. If the answer is "probably nothing" then maybe we should rephrase this to be more straight forward about that.

Reply:
opened issue about this https://github.com/pacu/zcash-swift-payment-uri/issues/34

Reply:
The intention of this is to state that this ZIP is intended to be backwards-compatible with such usage, that's all.